internal/config/validator.go

```diff
@@ -44,14 +44,35 @@ func (v *validator) Validate(resources ...client.ObjectList) error {
 			clusterResources.Pools = append(clusterResources.Pools, list.Items...)
 		case *metallbv1beta2.BGPPeerList:
 			clusterResources.Peers = append(clusterResources.Peers, list.Items...)
+			for i := range (clusterResources.Peers) {
+				clusterResources.Peers[i].Spec.BFDProfile = ""
+				clusterResources.Peers[i].Spec.PasswordSecret.Name = ""
+			}
 		case *metallbv1beta1.BFDProfileList:
 			clusterResources.BFDProfiles = append(clusterResources.BFDProfiles, list.Items...)
 		case *metallbv1beta1.BGPAdvertisementList:
 			clusterResources.BGPAdvs = append(clusterResources.BGPAdvs, list.Items...)
+			for i := range(clusterResources.BGPAdvs){
+				for j:= range(clusterResources.BGPAdvs[i].Spec.Communities){
+					if !existString(clusterResources.BGPAdvs[i].Spec.Communities[j],clusterResources.Communities){
+						clusterResources.BGPAdvs[i].Spec.Communities = removeElement(clusterResources.BGPAdvs[i].Spec.Communities, clusterResources.BGPAdvs[i].Spec.Communities[j])
+					}
+				}
+			}
 		case *metallbv1beta1.L2AdvertisementList:
 			clusterResources.L2Advs = append(clusterResources.L2Advs, list.Items...)
 		case *metallbv1beta1.AddressPoolList:
 			clusterResources.LegacyAddressPools = append(clusterResources.LegacyAddressPools, list.Items...)
+			for i := range (clusterResources.LegacyAddressPools) {
+				for j := range(clusterResources.LegacyAddressPools[i].Spec.BGPAdvertisements) {
+					for k := range(clusterResources.LegacyAddressPools[i].Spec.BGPAdvertisements[j].Communities){
+						if !existString(clusterResources.LegacyAddressPools[i].Spec.BGPAdvertisements[j].Communities[k], clusterResources.Communities) {
+							clusterResources.LegacyAddressPools[i].Spec.BGPAdvertisements[j].Communities = removeElement(clusterResources.LegacyAddressPools[i].Spec.BGPAdvertisements[j].Communities,
+								 clusterResources.LegacyAddressPools[i].Spec.BGPAdvertisements[j].Communities[k])
+						}
+					}
+				}
+			}	
```

Comment:
Using indexes to reference slice items make it very hard to read.
You can the second returned value (the object itself) from range as following:
```
for _, legacyAddrPool := range clusterResources.LegacyAddressPools {
    for _, bgpAdv := range legacyAddrPool {
    ...
    
```

Reply:
I agree and changed it.

Reply:
we still use indexes in existCommunity and resetTransientErrorsFields